<Type Name="SEHException" FullName="System.Runtime.InteropServices.SEHException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="20cce20d60f610a18cb52151a902d030a78984b2" /><Meta Name="ms.sourcegitcommit" Value="78161e70c57987ab89cc82ecd286618fd5b036c3" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="ru-RU" /><Meta Name="ms.lasthandoff" Value="10/10/2019" /><Meta Name="ms.locfileid" Value="72247068" /></Metadata><TypeSignature Language="C#" Value="public class SEHException : System.Runtime.InteropServices.ExternalException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit SEHException extends System.Runtime.InteropServices.ExternalException" />
  <TypeSignature Language="DocId" Value="T:System.Runtime.InteropServices.SEHException" />
  <TypeSignature Language="VB.NET" Value="Public Class SEHException&#xA;Inherits ExternalException" />
  <TypeSignature Language="C++ CLI" Value="public ref class SEHException : System::Runtime::InteropServices::ExternalException" />
  <TypeSignature Language="F#" Value="type SEHException = class&#xA;    inherit ExternalException" />
  <AssemblyInfo>
    <AssemblyName>System.Runtime.InteropServices</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.10.0</AssemblyVersion>
    <AssemblyVersion>4.0.20.0</AssemblyVersion>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.1.1.0</AssemblyVersion>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.1.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
    <BaseTypeName FrameworkAlternate="netframework-4.5">System.Runtime.InteropServices.ExternalException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netframework-4.8">
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
    <Attribute FrameworkAlternate="netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0">
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="f0c4b-101">Представляет ошибки структурной обработки исключений (SEH).</span><span class="sxs-lookup"><span data-stu-id="f0c4b-101">Represents structured exception handling (SEH) errors.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f0c4b-102">Класс <xref:System.Runtime.InteropServices.SEHException> обрабатывает ошибки SEH, вызываемые из неуправляемого кода, но не сопоставленные с другим исключением .NET Framework.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-102">The <xref:System.Runtime.InteropServices.SEHException> class handles SEH errors that are thrown from unmanaged code, but that have not been mapped to another .NET Framework exception.</span></span> <span data-ttu-id="f0c4b-103">Класс <xref:System.Runtime.InteropServices.SEHException> также соответствует `HRESULT` `E_FAIL` (0x80004005).</span><span class="sxs-lookup"><span data-stu-id="f0c4b-103">The <xref:System.Runtime.InteropServices.SEHException> class also corresponds to the `HRESULT` `E_FAIL` (0x80004005).</span></span>  
  
 <span data-ttu-id="f0c4b-104">.NET Framework часто сталкивается с неуправляемыми исключениями SEH, которые автоматически сопоставляются с управляемыми эквивалентами.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-104">The .NET Framework often encounters unmanaged SEH exceptions that are automatically mapped to managed equivalents.</span></span> <span data-ttu-id="f0c4b-105">Существует два распространенных исключения SEH:</span><span class="sxs-lookup"><span data-stu-id="f0c4b-105">There are two common unmanaged SEH exceptions:</span></span>  
  
-   <span data-ttu-id="f0c4b-106">исключения `STATUS_NO_MEMORY` автоматически сопоставляются с классом <xref:System.OutOfMemoryException>.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-106">`STATUS_NO_MEMORY` exceptions are automatically mapped to the <xref:System.OutOfMemoryException> class.</span></span>  
  
-   <span data-ttu-id="f0c4b-107">исключения `STATUS_ACCESS_VIOLATION` автоматически сопоставляются следующим образом:</span><span class="sxs-lookup"><span data-stu-id="f0c4b-107">`STATUS_ACCESS_VIOLATION` exceptions are automatically mapped as follows:</span></span>  
  
    -   <span data-ttu-id="f0c4b-108">Если применяется `legacyNullReferencePolicy`, все нарушения прав доступа сопоставляются с классом <xref:System.NullReferenceException>.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-108">If `legacyNullReferencePolicy` is applied, all access violations are mapped to the <xref:System.NullReferenceException> class.</span></span>  
  
    -   <span data-ttu-id="f0c4b-109">Если адрес, по которому была предпринята попытка чтения или записи, не находится в JIT-скомпилированном коде, это исключение сопоставляется с классом <xref:System.AccessViolationException>.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-109">If the address at which the read/write was attempted is not in JIT-compiled code, the exception is mapped to the <xref:System.AccessViolationException> class.</span></span>  
  
    -   <span data-ttu-id="f0c4b-110">Если адрес, на который была предпринята попытка чтения или записи, находится в JIT-скомпилированном коде, но отсутствует в области секции null, то исключение сопоставляется с классом <xref:System.AccessViolationException>.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-110">If the address at which the read/write was attempted is in JIT-compiled code, but it is not in the OS Null partition area, the exception is mapped to the <xref:System.AccessViolationException> class.</span></span>  
  
    -   <span data-ttu-id="f0c4b-111">Если `legacyNullReferencePolicy`, а адрес, по которому была предпринята попытка чтения или записи, находится в JIT-скомпилированном коде и в области секции null, исключение сопоставляется с классом <xref:System.NullReferenceException>.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-111">If there is no `legacyNullReferencePolicy`, and the address at which the read/write was attempted is in JIT-compiled code and in the OS Null partition area, the exception is mapped to the <xref:System.NullReferenceException> class.</span></span>  
  
 <span data-ttu-id="f0c4b-112">Любое исключение SEH, которое не было автоматически сопоставлено с конкретным исключением, по умолчанию сопоставляется с классом <xref:System.Runtime.InteropServices.SEHException>.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-112">Any SEH exception that is not automatically mapped to a specific exception is mapped to the <xref:System.Runtime.InteropServices.SEHException> class by default.</span></span>  
  
 <span data-ttu-id="f0c4b-113">Дополнительные сведения см. в разделе [Structured Exception Handling (C/C++)](/cpp/cpp/structured-exception-handling-c-cpp).</span><span class="sxs-lookup"><span data-stu-id="f0c4b-113">For more information, see [Structured Exception Handling (C/C++)](/cpp/cpp/structured-exception-handling-c-cpp).</span></span>  
  
 <span data-ttu-id="f0c4b-114">Обратите внимание, что класс <xref:System.Runtime.InteropServices.SEHException> не вызывает вызов деструкторов неуправляемого C++ исключения.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-114">Note that the <xref:System.Runtime.InteropServices.SEHException> class does not cause unmanaged C++ exception destructors to be called.</span></span> <span data-ttu-id="f0c4b-115">Чтобы обеспечить вызов деструкторов неуправляемого C++ исключения, используйте следующий синтаксис в блоке `catch`.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-115">To ensure that unmanaged C++ exception destructors are called, use the following syntax in the `catch` block.</span></span>  
  
```vb  
Catch   
     ' Handle catch here.  
End Try  
```  
  
```csharp  
catch  
{  
     // Handle catch here.  
}  
```  
  
```cpp  
catch(…)  
{  
     // Handle catch here.  
}   
```  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
    <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="f0c4b-116">Обработка и создание исключений в .NET</span><span class="sxs-lookup"><span data-stu-id="f0c4b-116">Handling and throwing exceptions in .NET</span></span></related>
    <related type="Article" href="~/docs/framework/interop/how-to-map-hresults-and-exceptions.md"><span data-ttu-id="f0c4b-117">Практическое руководство. Сопоставление значений HRESULT и исключений</span><span class="sxs-lookup"><span data-stu-id="f0c4b-117">How to: Map HRESULTs and Exceptions</span></span></related>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Runtime.InteropServices</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="f0c4b-118">Инициализирует новый экземпляр класса <see cref="T:System.Runtime.InteropServices.SEHException" />.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-118">Initializes a new instance of the <see cref="T:System.Runtime.InteropServices.SEHException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public SEHException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Runtime.InteropServices.SEHException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; SEHException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime.InteropServices</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="f0c4b-119">Инициализирует новый экземпляр класса <see cref="T:System.Runtime.InteropServices.SEHException" />.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-119">Initializes a new instance of the <see cref="T:System.Runtime.InteropServices.SEHException" /> class.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f0c4b-120">Класс <xref:System.Runtime.InteropServices.SEHException> наследуется от класса <xref:System.Runtime.InteropServices.ExternalException>.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-120">The <xref:System.Runtime.InteropServices.SEHException> class inherits from the <xref:System.Runtime.InteropServices.ExternalException> class.</span></span> <span data-ttu-id="f0c4b-121">Этот конструктор устанавливает свойства объекта <xref:System.Exception> в значения, указанные в следующей таблице.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-121">This constructor sets the properties of the <xref:System.Exception> object to the values in the following table.</span></span>  
  
|<span data-ttu-id="f0c4b-122">Свойство.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-122">Property</span></span>|<span data-ttu-id="f0c4b-123">Значение</span><span class="sxs-lookup"><span data-stu-id="f0c4b-123">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="f0c4b-124">Пустая ссылка (`Nothing` в Visual Basic).</span><span class="sxs-lookup"><span data-stu-id="f0c4b-124">A null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="f0c4b-125">Локализованная строка сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-125">A localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public SEHException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Runtime.InteropServices.SEHException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; SEHException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Runtime.InteropServices.SEHException : string -&gt; System.Runtime.InteropServices.SEHException" Usage="new System.Runtime.InteropServices.SEHException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime.InteropServices</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="f0c4b-126">Сообщение, указывающее причину возникновения исключения.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-126">The message that indicates the reason for the exception.</span></span></param>
        <summary><span data-ttu-id="f0c4b-127">Инициализирует новый экземпляр класса <see cref="T:System.Runtime.InteropServices.SEHException" /> с заданным сообщением.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-127">Initializes a new instance of the <see cref="T:System.Runtime.InteropServices.SEHException" /> class with a specified message.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f0c4b-128">Тип<xref:System.Runtime.InteropServices.SEHException> наследуется от типа <xref:System.Runtime.InteropServices.ExternalException>.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-128"><xref:System.Runtime.InteropServices.SEHException> inherits from <xref:System.Runtime.InteropServices.ExternalException>.</span></span> <span data-ttu-id="f0c4b-129">Этот конструктор задает свойства объекта <xref:System.Exception>, как показано в следующей таблице.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-129">This constructor sets the properties of the <xref:System.Exception> object as shown in the following table.</span></span>  
  
|<span data-ttu-id="f0c4b-130">Свойство.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-130">Property</span></span>|<span data-ttu-id="f0c4b-131">Значение</span><span class="sxs-lookup"><span data-stu-id="f0c4b-131">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="f0c4b-132">Пустая ссылка (`Nothing` в Visual Basic).</span><span class="sxs-lookup"><span data-stu-id="f0c4b-132">A null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="f0c4b-133">`message`.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-133">`message`.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected SEHException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Runtime.InteropServices.SEHException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; SEHException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new System.Runtime.InteropServices.SEHException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; System.Runtime.InteropServices.SEHException" Usage="new System.Runtime.InteropServices.SEHException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime.InteropServices</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Security.SecuritySafeCritical</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" Index="0" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;netstandard-2.0;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netstandard-2.1" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" Index="1" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;netstandard-2.0;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netstandard-2.1" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="f0c4b-134">Объект, содержащий сериализованные данные объекта.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-134">The object that holds the serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="f0c4b-135">Контекстные сведения об источнике или назначении.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-135">The contextual information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="f0c4b-136">Инициализирует новый экземпляр класса <see cref="T:System.Runtime.InteropServices.SEHException" />, используя данные сериализации.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-136">Initializes a new instance of the <see cref="T:System.Runtime.InteropServices.SEHException" /> class from serialization data.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f0c4b-137">Данный конструктор вызывается в процессе десериализации для воссоздания объекта исключения, переданного в потоке.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-137">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="f0c4b-138">Дополнительные сведения см. в описании пространства имен <xref:System.Runtime.Serialization>.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-138">For more information, see the <xref:System.Runtime.Serialization> namespace.</span></span>  
  
 <span data-ttu-id="f0c4b-139">Тип<xref:System.Runtime.InteropServices.SEHException> наследуется от типа <xref:System.Runtime.InteropServices.ExternalException>.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-139"><xref:System.Runtime.InteropServices.SEHException> inherits from <xref:System.Runtime.InteropServices.ExternalException>.</span></span>  
  
 ]]></format>
        </remarks>
        <exception cref="T:System.ArgumentNullException"><span data-ttu-id="f0c4b-140">Свойство <paramref name="info" /> имеет значение <see langword="null" />.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-140"><paramref name="info" /> is <see langword="null" />.</span></span></exception>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public SEHException (string message, Exception inner);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception inner) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Runtime.InteropServices.SEHException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, inner As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; SEHException(System::String ^ message, Exception ^ inner);" />
      <MemberSignature Language="F#" Value="new System.Runtime.InteropServices.SEHException : string * Exception -&gt; System.Runtime.InteropServices.SEHException" Usage="new System.Runtime.InteropServices.SEHException (message, inner)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime.InteropServices</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="inner" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="f0c4b-141">Сообщение об ошибке, указывающее причину создания исключения.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-141">The error message that explains the reason for the exception.</span></span></param>
        <param name="inner"><span data-ttu-id="f0c4b-142">Исключение, которое является причиной текущего исключения.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-142">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="f0c4b-143">Если значение параметра <paramref name="inner" /> не равно <see langword="null" />, текущее исключение сгенерировано в блоке <see langword="catch" />, обрабатывающем внутреннее исключение.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-143">If the <paramref name="inner" /> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="f0c4b-144">Инициализирует новый экземпляр класса <see cref="T:System.Runtime.InteropServices.SEHException" /> указанным сообщением об ошибке и ссылкой на внутреннее исключение, вызвавшее данное исключение.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-144">Initializes a new instance of the <see cref="T:System.Runtime.InteropServices.SEHException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f0c4b-145">Исключение, созданное как прямой результат предыдущего исключения, должно содержать в свойстве <xref:System.Exception.InnerException%2A> ссылку на предыдущее исключение.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-145">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="f0c4b-146">Свойство <xref:System.Exception.InnerException%2A> возвращает значение, переданное конструктору, или значение `null`, если свойство <xref:System.Exception.InnerException%2A> не передает конструктору значение внутреннего исключения.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-146">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="f0c4b-147">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.Runtime.InteropServices.SEHException>.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-147">The following table shows the initial property values for an instance of <xref:System.Runtime.InteropServices.SEHException>.</span></span>  
  
|<span data-ttu-id="f0c4b-148">Свойство.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-148">Property</span></span>|<span data-ttu-id="f0c4b-149">Значение</span><span class="sxs-lookup"><span data-stu-id="f0c4b-149">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="f0c4b-150">Ссылка на внутреннее исключение.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-150">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="f0c4b-151">Строка сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-151">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="f0c4b-152">Обработка и создание исключений в .NET</span><span class="sxs-lookup"><span data-stu-id="f0c4b-152">Handling and throwing exceptions in .NET</span></span></related>
      </Docs>
    </Member>
    <Member MemberName="CanResume">
      <MemberSignature Language="C#" Value="public virtual bool CanResume ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig newslot virtual instance bool CanResume() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Runtime.InteropServices.SEHException.CanResume" />
      <MemberSignature Language="VB.NET" Value="Public Overridable Function CanResume () As Boolean" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; virtual bool CanResume();" />
      <MemberSignature Language="F#" Value="abstract member CanResume : unit -&gt; bool&#xA;override this.CanResume : unit -&gt; bool" Usage="sEHException.CanResume " />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime.InteropServices</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Boolean</ReturnType>
      </ReturnValue>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="f0c4b-153">Показывает, возможно ли восстановление после исключения и возможно ли продолжить выполнение кода с точки возникновения этого исключения.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-153">Indicates whether the exception can be recovered from, and whether the code can continue from the point at which the exception was thrown.</span></span></summary>
        <returns><span data-ttu-id="f0c4b-154">Всегда имеет значение <see langword="false" />, поскольку исключения с возможностью восстановления не реализованы.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-154">Always <see langword="false" />, because resumable exceptions are not implemented.</span></span></returns>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f0c4b-155">Если <xref:System.Runtime.InteropServices.SEHException.CanResume%2A> возвращает `true`, то отфильтрованный обработчик исключений может устранить проблему, вызвавшую исключение, и код продолжит работу с момента возникновения исключения.</span><span class="sxs-lookup"><span data-stu-id="f0c4b-155">If <xref:System.Runtime.InteropServices.SEHException.CanResume%2A> returns `true`, then a filtered exception handler can correct the problem that caused the exception, and the code will continue from the point at which the exception was thrown.</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>
